### An important quality characteristic used by the manufacturers of ABC asphalt shingles is the amount of moisture the shingles contain when they are packaged. Customers may feel that they have purchased a product lacking in quality if they find moisture and wet shingles inside the packaging.   In some cases, excessive moisture can cause the granules attached to the shingles for texture and colouring purposes to fall off the shingles resulting in appearance problems. To monitor the amount of moisture present, the company conducts moisture tests. A shingle is weighed and then dried. The shingle is then reweighed, and based on the amount of moisture taken out of the product, the pounds of moisture per 100 square feet are calculated. The company would like to show that the mean moisture content is less than 0.35 pound per 100 square feet.

### The file (A & B shingles.csv) includes 36 measurements (in pounds per 100 square feet) for A shingles and 31 for B shingles.

3.1 Do you think there is evidence that mean moisture contents in both types of shingles are within the permissible limits? State your conclusions clearly showing all steps.

3.2 Do you think that the population means for shingles A and B are equal? Form the hypothesis and conduct the test of the hypothesis. What assumption do you need to check before the test for equality of means is performed?

In [1]:
import os
os.chdir('C:\\Users\\suhas\\desktop\\Great Learning\\2.STATISTICS\\FinalAssignment_STATISTICS')
os.getcwd()

'C:\\Users\\suhas\\desktop\\Great Learning\\2.STATISTICS\\FinalAssignment_STATISTICS'

In [2]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
import scipy.stats as stats
from scipy.stats import ttest_1samp, ttest_ind
from statsmodels.stats.power import ttest_power
from scipy.stats import f

In [3]:
df = pd.read_csv('A+&+B+shingles.csv')
df

A     B
0   0.44  0.14
1   0.61  0.15
2   0.47  0.31
3   0.30  0.16
4   0.15  0.37
5   0.24  0.18
6   0.16  0.42
7   0.20  0.58
8   0.20  0.25
9   0.20  0.41
10  0.26  0.17
11  0.14  0.13
12  0.33  0.23
13  0.13  0.11
14  0.72  0.10
15  0.51  0.19
16  0.28  0.22
17  0.39  0.44
18  0.39  0.11
19  0.25  0.11
20  0.16  0.31
21  0.20  0.43
22  0.22  0.26
23  0.42  0.18
24  0.24  0.44
25  0.21  0.43
26  0.49  0.16
27  0.34  0.52
28  0.36  0.36
29  0.29  0.22
30  0.27  0.39
31  0.40   NaN
32  0.29   NaN
33  0.43   NaN
34  0.34   NaN
35  0.37   NaN

##### 3.1 Do you think there is evidence that mean moisture contents in both types of shingles are within the permissible limits? State your conclusions clearly showing all steps.

In [4]:
df['A']

0     0.44
1     0.61
2     0.47
3     0.30
4     0.15
5     0.24
6     0.16
7     0.20
8     0.20
9     0.20
10    0.26
11    0.14
12    0.33
13    0.13
14    0.72
15    0.51
16    0.28
17    0.39
18    0.39
19    0.25
20    0.16
21    0.20
22    0.22
23    0.42
24    0.24
25    0.21
26    0.49
27    0.34
28    0.36
29    0.29
30    0.27
31    0.40
32    0.29
33    0.43
34    0.34
35    0.37
Name: A, dtype: float64

H0 : mean moisture content <=0.35
HA : mean moisture content > 0.35

alpha = 0.05

In [5]:
# We are using t test 

In [6]:
t1,p1_value = stats.ttest_1samp(df['A'],0.35)
print(t1,p1_value)

-1.4735046253382782 0.14955266289815025


In [7]:
## p-value for a one-tailed test, we have to divide it by 2

p1_value

print ('The T statistic is:',t1,'\n''The corresponding pvalue is :',(p1_value/2))

The T statistic is: -1.4735046253382782 
The corresponding pvalue is : 0.07477633144907513


In [8]:
# For Shingles B
t2,p2_value = stats.ttest_1samp(df['B'],0.35,nan_policy='omit')
print(t2,p2_value)


-3.1003313069986995 0.004180954800638365


In [9]:
print ('The T statistic is:',t2,'\n''The corresponding pvalue is :',(p2_value/2))

The T statistic is: -3.1003313069986995 
The corresponding pvalue is : 0.0020904774003191826


For Shingles A value of p is 0.07477633144907513 is greater than alpha, 0.07477633144907513 > 0.05, Null Hypothesis is accepted. mean moisture contents in type A of shingles are within the permissible limits. 

For Shingles B value of p is 0.07477633144907513 is less than alpha, 0.0020904774003191826  < 0.05, Null Hypothesis will be rejected.mean moisture contents in type B of shingles are not within the permissible limits. to get more correct results we need more no of samples.

##### 3.2 Do you think that the population means for shingles A and B are equal? Form the hypothesis and conduct the test of the hypothesis. What assumption do you need to check before the test for equality of means is performed?


H0 : Population mean of Shingles A = Population mean of Shingles B

HA : Population mean of Shingles A greater than or less than Population mean of Shingles B

alpha= 0.05

In [10]:
# We are using 2 sample t test

alpha = 0.05

t3, p3_value = ttest_ind(df['A'], df['B'], equal_var=True,nan_policy='omit')
print(t3,p3_value)

1.2896282719661123 0.2017496571835306


In [11]:
print ('The T statistic is:',t3,'\n''The corresponding pvalue is :',(p3_value))

The T statistic is: 1.2896282719661123 
The corresponding pvalue is : 0.2017496571835306


In [12]:
if p3_value > alpha :
    print('We fail to Reject the Null Hypothesis')
else:
    print('We Reject the Null Hypothesis')

We fail to Reject the Null Hypothesis


Here, P value is 0.2017496571835306 is greater than 0.05 hence Null hypothesis will be accepted

##### Two-Sample T-Test Assumptions we made:
The assumptions of the two-sample t-test are: The data are continuous (not discrete).

The data follow the normal probability distribution.

The two samples are independent.

Both samples are simple random samples from their respective populations.
